### PR TITLE
feat: add header-tailwind-ui

### LINKED ISSUE #122: [FEATURE] <Create a header bar with Tailwind UI>
### Description

We can probably just use [simple dark](https://tailwindui.com/components/application-ui/navigation/navbars#component-1f0fd3139f66c75457637ca3746e3384).

Create a header bar with a login/signup link.

If the user is already logged in, we should show the user's info - (part of) the email address for now.

### PR DESCRIPTION
**Pull request by omkarg01**
### 🛠️ Fixes Issue
closes #122

### 👨‍💻 Changes proposed
Added Header of Tailwind Ui.
Added dependencies: -
'@headlessui/react'
'@heroicons/react/outline'

### 📷 Screenshots
heaeder added

### CODE REVIEW

**Comment by ykdojo** [comment]
Looks good! Is it ready for a review?

**Comment by omkarg01** [comment]
> Looks good! Is it ready for a review?

Yes

File: src/components/Header.tsx [hunk 1]
```diff
@@ -1,31 +1,164 @@
...
+								href='#'
+								className='block rounded-md px-3 py-2 text-base font-medium text-gray-300 hover:bg-gray-700 hover:text-white'
+							>
+								Calendar
+							</Disclosure.Button>
+						</div>
+						<div className='border-t border-gray-700 pt-4 pb-3'>
+							<div className='flex items-center px-5'>
+								<div className='flex-shrink-0'>
+									<img
```
Comment (ykdojo) [inline src/components/Header.tsx:117]: we don't have a profile image implemented yet.

can you use a username / email / part of an email instead?

**Review by ykdojo — CHANGES_REQUESTED** [review]

File: package.json [hunk 2]
```diff
@@ -11,6 +11,8 @@
 		"prepare": "husky install"
 	},
 	"dependencies": {
+		"@headlessui/react": "^1.6.6",
+		"@heroicons/react": "^1.0.6",
```
Comment (iShibi) [inline package.json:15]: You can remove this dep and import herocions from the already installed `react-icons` package.

File: src/components/Header.tsx [hunk 3]
```diff
@@ -1,31 +1,164 @@
+/* This example requires Tailwind CSS v2.0+ */
+import { Fragment } from 'react';
+import { Disclosure, Menu, Transition } from '@headlessui/react';
+import { BellIcon, MenuIcon, XIcon } from '@heroicons/react/outline';
```
Comment (iShibi) [inline src/components/Header.tsx:4]: Import these icons from `react-icons`. You can find their names here https://react-icons.github.io/react-icons/icons?name=hi
```suggestion
import { HiBell, HiMenu, HiX } from 'react-icons/hi';
```

File: src/components/Header.tsx [hunk 4]
```diff
@@ -1,31 +1,164 @@
+/* This example requires Tailwind CSS v2.0+ */
```
Comment (iShibi) [inline src/components/Header.tsx:1]: You can remove this comment
```suggestion
```

File: src/components/Header.tsx [hunk 5]
```diff
@@ -1,31 +1,164 @@
...
+										</a>
+									</Link>
+								</div>
+							</div>
+							<div className='-mr-2 flex sm:hidden'>
+								{/* Mobile menu button */}
+								<Disclosure.Button className='inline-flex items-center justify-center rounded-md p-2 text-gray-400 hover:bg-gray-700 hover:text-white focus:outline-none focus:ring-2 focus:ring-inset focus:ring-white'>
+									<span className='sr-only'>Open main menu</span>
+									{open ? (
+										<XIcon className='block h-6 w-6' aria-hidden='true' />
```
Comment (iShibi) [inline src/components/Header.tsx:73]:
```suggestion
										<HiX className='block h-6 w-6' aria-hidden='true' />
```

File: src/components/Header.tsx [hunk 6]
```diff
@@ -1,31 +1,164 @@
...
+									) : (
+										<MenuIcon className='block h-6 w-6' aria-hidden='true' />
```
Comment (iShibi) [inline src/components/Header.tsx:75]:
```suggestion
										<HiMenu className='block h-6 w-6' aria-hidden='true' />
```

File: src/components/Header.tsx [hunk 7]
```diff
@@ -1,31 +1,164 @@
...
+											className='rounded-md px-3 py-2 text-sm font-medium text-gray-300 hover:bg-gray-700 hover:text-white'
+										>
+											Calendar
+										</a>
+									</div>
+								</div>
+							</div>
+							<div className='hidden sm:ml-6 sm:block'>
+								<div className='flex items-center'>
+									<Link href='#'>
```
Comment (iShibi) [inline src/components/Header.tsx:61]:
```suggestion
									<Link href='/signin'>
```

File: src/components/Header.tsx [hunk 8]
```diff
@@ -1,31 +1,164 @@
+/* This example requires Tailwind CSS v2.0+ */
+import { Fragment } from 'react';
+import { Disclosure, Menu, Transition } from '@headlessui/react';
+import { BellIcon, MenuIcon, XIcon } from '@heroicons/react/outline';
+import SignIn from '../pages/signin';
```
Comment (iShibi) [inline src/components/Header.tsx:5]: Why import a page in a component?

**Review by iShibi — CHANGES_REQUESTED** [review]

**Comment by subhoghoshX** [comment]
Edited the PR description to link it to issue #122

File: src/components/Header.tsx (hunk 1 above)
Comment (omkarg01) [reply src/components/Header.tsx:117]: > we don't have a profile image implemented yet.
> 
> can you use a username / email / part of an email instead?

no_profimage3


@ykdojo you mean removing the profile image and keeping the email as this?

**Comment by iShibi** [comment]
@omkarg01 Do not resolve conversations without applying the suggested changes.

File: src/components/Header.tsx (hunk 1 above)
Comment (subhoghoshX) [reply src/components/Header.tsx:117]: @ykdojo why not keep it there as a placeholder for now. I think we're going to implement it anyway.

Comment (ykdojo) [reply src/components/Header.tsx]: I'm ok either way!

**Comment by omkarg01** [comment]
@ykdojo @iShibi review the changes and let me know any further improvements I can make to it

File: src/components/Header.tsx [hunk 9]
```diff
@@ -1,31 +1,152 @@
...
+										>
+											Calendar
+										</a>
+									</div>
+								</div>
+							</div>
+							<div className='hidden sm:ml-6 sm:block'>
+								<div className='flex items-center'>
+									<Link href='/signin'>
+										<a className='text-white' target='_blank' rel='noopener noreferrer'>
```
Comment (iShibi) [inline src/components/Header.tsx:58]: This shouldn't open in a new tab. Also, there is no need for a `rel` attribute here as it's an internal link.
```suggestion
										<a className='text-white'>
```

**Review by iShibi — CHANGES_REQUESTED** [review]

**Comment by iShibi** [comment]
Some part of this UI template seems to be missing. It shows `Your Profile`, `Sign Out`, and `Settings` options in the mobile view but they are not there in the desktop view.

**Comment by omkarg01** [comment]
> Some part of this UI template seems to be missing. It shows `Your Profile`, `Sign Out`, and `Settings` options in the mobile view but they are not there in the desktop view.

Looks perfect now?

**Comment by omkarg01** [comment]
> Some part of this UI template seems to be missing. It shows `Your Profile`, `Sign Out`, and `Settings` options in the mobile view but they are not there in the desktop view.

@iShibi Looks perfect now?

**Comment by iShibi** [comment]
@omkarg01 Your lockfile is out of sync. Delete the `package-lock.json` and run `npm i` to sync it.

**Comment by omkarg01** [comment]
@iShibi why is there a merge conflict? I need help, understanding this issue.

**Review by rohitdasu — APPROVED** [review]
LGTM 👍

**Pull request merged**